Code/Framework/AzCore/AzCore/variadic.h

```diff
@@ -62,3 +76,300 @@
+#define AZ_FOR_EACH_114(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_115(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_116(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_117(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_118(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_119(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_120(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_121(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_122(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_123(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_124(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+#define AZ_FOR_EACH_125(pppredicate, separator, param, ...) pppredicate(param) separator AZ_FOR_EACH(pppredicate, __VA_ARGS__)
+
+
+// Invokes unary predicate macro on each variadic argument, without any separator between calls to unary predicate
+#define AZ_FOR_EACH(pp_unary_predicate, ...) AZ_MACRO_CALL(AZ_JOIN(AZ_FOR_EACH_, AZ_VA_NUM_ARGS(__VA_ARGS__)),pp_unary_predicate,,__VA_ARGS__)
+// Invokes unary predicate macro on each variadic argument with a separator in between
+#define AZ_FOR_EACH_WITH_SEPARATOR(pp_unary_predicate, ...) AZ_MACRO_CALL(AZ_JOIN(AZ_FOR_EACH_, AZ_VA_NUM_ARGS(__VA_ARGS__)),pp_unary_predicate,,__VA_ARGS__)
+
+// Unpacks a (...) parameter into variable arguments
+// Match case when a "()" __VA_ARGS__ is unwrapped
+#define AZ_FOR_EACH_UNWRAP_UNPACKED(pp_unary_predicate, ...) AZ_FOR_EACH(pp_unary_predicate,__VA_ARGS__)
+#define AZ_FOR_EACH_UNWRAP(pp_unary_predicate, parenthesized_arg) AZ_FOR_EACH_UNWRAP_UNPACKED(pp_unary_predicate,AZ_UNWRAP(parenthesized_arg))
+// Unwraps a parenthesize expression and invokes AZ_FOR_EACH_WITH_SEPARATOR
+#define AZ_FOR_EACH_UNWRAP_UNPACKED_WITH_SEPARATOR(pp_unary_predicate, separtor, ...) AZ_FOR_EACH_WITH_SEPARATOR(pp_unary_predicate,separator,__VA_ARGS__)
+#define AZ_FOR_EACH_UNWRAP_WITH_SEPARATOR(pp_unary_predicate, separator, parenthesized_arg) AZ_FOR_EACH_UNWRAP_UNPACKED_WITH_SEPARATOR(pp_unary_predicate,separator,AZ_UNWRAP(parenthesized_arg))
+
+#define AZ_REMOVE_PARENTHESIS(...) AZ_REMOVE_PARENTHESIS __VA_ARGS__
```

Comment:
Is this used?  It uses itself in its definition, does that even work?

Reply:
Actually yes.
It recursively removes parenthesis by resolving the macro of "AZ_REMOVE_PARENTHESIS (())"
If the there is no parenthesis the macro resolve AZ_REMOVE_PARENTHESIS resolves to nothing since it is not being called